### PR TITLE
Minor refactor

### PR DESCRIPTION
**Pull request by zekth**
This refactor a little sketchy pattern.

### CODE REVIEW

File: lib/registry.js [hunk 1]
```diff
@@ -43,14 +43,14 @@ class Registry {
...
+			const labels = [];
 			for (const key of Object.keys(val.labels)) {
-				labels += `${key}="${escapeLabelValue(val.labels[key])}",`;
+				labels.push(`${key}="${escapeLabelValue(val.labels[key])}"`);
 			}
 
 			let metricName = val.metricName || item.name;
 			if (labels) {
-				metricName += `{${labels.substring(0, labels.length - 1)}}`;
+				metricName += `{${labels.join(',')}}`;
```
Comment (zbjornson) [inline lib/registry.js]: Thanks for the PRs!

Unfortunately this approach is slower:

```
✗ registry ➭ metrics#1 with 64 is 11.13% slower.
⚠ registry ➭ metrics#2 with 8 is 7.170% acceptably slower.
⚠ registry ➭ metrics#2 with 4 and 2 with 2 is 0.4725% acceptably slower.
⚠ registry ➭ metrics#2 with 2 and 2 with 4 is 8.324% acceptably slower.
⚠ registry ➭ metrics#6 with 2 is 6.719% acceptably slower.
```

If you want to change it, could you try out the approach outlined in https://github.com/siimon/prom-client/pull/235#discussion_r244376041? You can run `yarn/npm benchmarks` while on your branch to compare against the current release.

**Comment by zekth** [comment]
Update:
```
✓ registry ➭ getMetricsAsJSON#1 with 64 is 1.652% faster.
✓ registry ➭ getMetricsAsJSON#2 with 8 is 0.8094% faster.
✓ registry ➭ getMetricsAsJSON#2 with 4 and 2 with 2 is 4.417% faster.
⚠ registry ➭ getMetricsAsJSON#2 with 2 and 2 with 4 is 2.478% acceptably slower.
⚠ registry ➭ getMetricsAsJSON#6 with 2 is 1.328% acceptably slower.
✓ registry ➭ metrics#1 with 64 is 13.89% faster.
✓ registry ➭ metrics#2 with 8 is 8.079% faster.
✓ registry ➭ metrics#2 with 4 and 2 with 2 is 7.516% faster.
✓ registry ➭ metrics#2 with 2 and 2 with 4 is 9.523% faster.
⚠ registry ➭ metrics#6 with 2 is 0.7138% acceptably slower.
```

**Comment by siimon** [comment]
I don't really see the point, sorry.. Could you give me some insight on why you think this refactor is needed?

**Comment by zekth** [comment]
> I don't really see the point, sorry.. Could you give me some insight on why you think this refactor is needed?

At first it was only to tackle the `substring` pattern, but as it was slower just changed the implementation like suggested and improved the perf.

File: lib/registry.js [hunk 2]
```diff
@@ -43,14 +43,20 @@ class Registry {
...
+			const keys = Object.keys(val.labels);
+			const size = keys.length;
+			if (size > 0) {
+				let labels = '';
+				let i = 0;
+				for (; i < size; i++) {
+					labels += `${keys[i]}="${escapeLabelValue(val.labels[keys[i]])}"`;
+					if (i !== size - 1) {
+						labels += ',';
+					}
```
Comment (zbjornson) [inline lib/registry.js]: Almost. The exact way I like to do this is to loop up to `size - 1` and not have an `if` in the loop body (see comment I linked to previously). I think that will save a line also.

**Comment by siimon** [comment]
Alright, could you update the changelog please?

File: CHANGELOG.md [hunk 3]
```diff
@@ -45,7 +46,7 @@ project adheres to [Semantic Versioning](http://semver.org/).
 ### Added
 
 - feat: implement GC metrics collection without native(C++) modules.
-- faet: implement advanced event loop monitoring
+- feat: implement advanced event loop monitoring
```
Comment (zekth) [inline CHANGELOG.md]: Fixed this typo also

**Pull request merged**